test/annexB/language/function-code/function-named-arguments-hoists.js

```diff
@@ -0,0 +1,35 @@
+/*---
+description: Functions named 'arguments' have legacy hoisting semantics
+esid: sec-web-compat-functiondeclarationinstantiation
+flags: [noStrict]
+info: |
+    FunctionDeclarationInstantiation ( _func_, _argumentsList_ )
+
+    [...]
+    1. Let _parameterNames_ be the BoundNames of _formals_.
+    [...]
+
+    Changes to FunctionDeclarationInstantiation
+
+    [...]
+    1. If replacing the |FunctionDeclaration| _f_ with a |VariableStatement| that has _F_ as a |BindingIdentifier| would not produce any Early Errors for _func_ and _F_ is not an element of _parameterNames_, then
+      [...]
+---*/
+
+// Simple parameters
+(function() {
```

Comment:
It might be worth adding an extra case to differentiate empty from a named simple parameter